torch/_inductor/codegen/cpp.py

```diff
@@ -3260,6 +3206,124 @@ def select_tiling(
         )
         if tiling_indices:
+            if config.cpp.enable_tiling_heuristics:
+
+                def _try_get_stride(
+                    index,
+                    itervars,
+                    tiling_factor,
+                    tiling_indices,
+                ):
+                    itervar = itervars[tiling_indices[0]]
+                    stride = stride_at_vec_range(index, itervar, tiling_factor)
+                    return stride if stride.is_number else None
+
+                def _update_negative_op_count(
+                    node_name, non_contig_indexing_op_counter
+                ):
+                    if node_name not in non_contig_indexing_op_counter:
+                        non_contig_indexing_op_counter[node_name] = 1
+                    else:
+                        non_contig_indexing_op_counter[node_name] += 1
+
+                def _is_valid_indices(
+                    itervars,
+                    tiling_indices,
+                ):
+                    return (
+                        len(tiling_indices) == 1
+                        and len(itervars) > 0
+                        and (
+                            tiling_indices[0]
+                            if tiling_indices[0] >= 0
+                            else tiling_indices[0] + len(itervars)
+                        )
+                        < len(itervars)
+                    )
+
+                group, reduction_group = max(
+                    var_sizes_list, key=lambda sizes: len(sizes[1])
+                )
+                call_ranges = tuple(group) + tuple(reduction_group)
+                itervars = [
+                    sympy_index_symbol_with_prefix(SymT.XBLOCK, n)
+                    for n in range(len(call_ranges))
+                ]
+                reduction_depth = len(group)
+                vars, reduction_vars = (
+                    itervars[:reduction_depth],
+                    itervars[reduction_depth:],
+                )
+                op_counter: Dict[str, int] = {}
+                # ops may cause overhead with vectorization, like non-contiguous
+                # index_expr, load, store
+                non_contig_indexing_op_counter: Dict[str, int] = {}
+                for _body in loop_bodies:
+                    sub_blocks = [_body.root_block] + list(_body.subblocks.values())
+                    for sub_block in sub_blocks:
+                        for _node in sub_block.graph.nodes:
+                            if _node.target in ["index_expr", "load", "store"]:
+                                # get the index and replace prefix from z to x
+                                index = sub_block.body.indexing_from_args(
+                                    (vars, reduction_vars)
+                                )[
+                                    _node.args[
+                                        1 if _node.target == "index_expr" else 2
+                                    ].args[0]
```

Comment:
@leslie-fang-intel this is the line that raises on the internal workload

Reply:
Thanks, will try to submit fix. Still not sure yet, how could the `_node.args[1 if _node.target == "index_expr" else 2]` could be a str, but I guess we can bypass this special case.

Reply:
I'll keep debugging on my side; unfortunately the failing test takes a while to run :/

Reply:
Thanks, here we just want to get the name of index expression. So for all of the cases we see in the 3 test suits (TorchBench, Timms, Huggingface) we expect, take `index_expr` as example:

- `_node` is a FX Node with target `index_expr`
- `_node.args[1 if _node.target == "index_expr" else 2]` is another FX node with target `get_index`
- `_node.args[1 if _node.target == "index_expr" else 2].args[0]` is a str for the name of this index expression

For your case, it seems `_node.args[1 if _node.target == "index_expr" else 2]` is a str instead of the FX node.

Reply:
@leslie-fang-intel Sorry for the delay. I was trying to debug last night and I'm seeing some extremely strange behavior I can't explain. I thought I might be hallucinating and just needed sleep. When I poke around in the debugger, the element in question is not a string. Amazingly, when I change the code to the following, there is no error and the test runs successfully:

```
                            iii = 1 if _node.target == "index_expr" else 2
                            _node.args[
                                iii
                            ].args[0]
```

What?!